### ISSUE TITLE
Can not build: JavaLocatorError OS X

### ISSUE THREAD

**Issue by anthonyjchriste** [astonbitecode/j4rs#6]
I am having trouble building with  j4rs and get the following error:

`Error: J4rsBuildError { description: "JavaLocatorError { description: \"Could not find the libjvm.* library in any subdirectory of /System/Library/Frameworks/JavaVM.framework/Versions/Current\" }" }`

System Details:
* OS X Mojave 10.14.5
* JDK 1.8.0_152
* cargo 1.35.0 (6f3e9c367 2019-04-04)
* j4rs 0.6.0

Contents of above directory appears to be a symbolic link

```
> ls -l /System/Library/Frameworks/JavaVM.framework/Versions/Current
lrwxr-xr-x  1 root  wheel  1 Mar  5 10:26 /System/Library/Frameworks/JavaVM.framework/Versions/Current -> A
```

Contents of sym link:

```
> ls -l /System/Library/Frameworks/JavaVM.framework/Versions/A
total 80
drwxr-xr-x  52 root  wheel    1664 Jun  4 05:37 Commands
drwxr-xr-x   4 root  wheel     128 Sep 20  2018 Frameworks
drwxr-xr-x   3 root  wheel      96 Aug 17  2018 JavaPluginCocoa.bundle
-rwxr-xr-x   1 root  wheel  107984 May  3 20:53 JavaVM
drwxr-xr-x  50 root  wheel    1600 Jun  4 05:37 Resources
drwxr-xr-x   3 root  wheel      96 Jun  4 05:37 _CodeSignature
```

This error is new and only started appearing since I've updated OS X and the JDK through homebrew. 

Please let me know if I can provide any other details or perform any additional steps to help debug this issue. 

I appreciate your time.

**Comment by astonbitecode** [comment]
Please set your `JAVA_HOME` env var to point to your Java installation in order to bypass the problem. I will try to reproduce the issue @Jenkins (no OS X available locally unfortunately)

**Comment by anthonyjchriste** [comment]
Thank you! Setting JAVA_HOME does the trick. If I wanted to debug and submit a patch for this issue, where would I look? In your crate or one of the dependent crates?

**Comment by astonbitecode** [comment]
Thanks, I appreciate your help! 

The issue is in the [java-locator](https://github.com/astonbitecode/java-locator) crate. This is a small utility that takes the burden from the j4rs user, finding the Java installation automatically. This is needed mostly for finding the `jvm` dynamic library.

[Here](https://github.com/astonbitecode/j4rs/blob/135d2288f692712b1e6f2fd6a655332500411b5e/rust/src/api_tweaks/generic.rs#L39) should be your starting point (calls to `locate_jvm_dyn_library` and `get_jvm_dyn_lib_file_name`).

**Comment by anthonyjchriste** [comment]
Great, thank you! I will spend some time with this and hopefully have a PR for you in the future!

**Comment by anthonyjchriste** [comment]
So the locations of libjvm.dylib are in:

`
/Library/Java/JavaVirtualMachines/jdk-12.0.1.jdk/Contents/Home/lib/server/libjvm.dylib
/Library/Java/JavaVirtualMachines/jdk1.8.0_152.jdk/Contents/Home/jre/lib/server/libjvm.dylib
`

However the java binary points to: `lrwxr-xr-x  1 root  wheel  74 Mar  5 10:30 /usr/bin/java -> /System/Library/Frameworks/JavaVM.framework/Versions/Current/Commands/java`

I'm not sure if there is an easy way to lookup "JAVA_HOME" by "whiching" the binary.

I'll keep looking for a generic solution.

**Comment by astonbitecode** [comment]
Assuming that the issue is for jdk 12 and according to [this](https://docs.oracle.com/en/java/javase/12/install/installed-directory-structure-jdk.html#GUID-F7178F2F-DC92-47E9-8062-CA6B2612D350), I would expect that the binary should be in a directory like `/jdk-12.0.1.jdk/bin`.

> However the java binary points to: lrwxr-xr-x 1 root wheel 74 Mar 5 10:30 /usr/bin/java -> /System/Library/Frameworks/JavaVM.framework/Versions/Current/Commands/java

Does maybe `/System/Library/Frameworks/JavaVM.framework/Versions/Current/Commands/java` point somewhere else?

**Comment by astonbitecode** [comment]
Fixed in the java_locator [0.1.1](https://crates.io/crates/java-locator).

Thanks for reporting.
